### PR TITLE
Fix for missing api-version in OperationLocation

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Resolve https://github.com/Azure/azure-sdk-for-net/issues/25411

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/assets/Generator.Shared/NextLinkOperationImplementation.cs [hunk 1]
```diff
@@ -26,6 +26,7 @@ internal class NextLinkOperationImplementation : IOperation
 
         private string? _lastKnownLocation;
         private string _nextRequestUri;
+        private string? _apiVersion;
```
Comment (annelo-msft) [inline src/assets/Generator.Shared/NextLinkOperationImplementation.cs:29]: I'm wondering how this approach will scale to accommodate all of the customizations that could happen in code generated from a swagger file.  For example, there is a recent customer request to handle a 207 response as a request as well - would we handle that here, too?  Or is there a way to add an abstract method (or something similar) to call out to code generated for the library to create the message for polling requests?

File: src/assets/Generator.Shared/NextLinkOperationImplementation.cs [hunk 2]
```diff
@@ -115,6 +115,17 @@ private void UpdateNextRequestUri(ResponseHeaders headers)
             }
         }
 
+        private static string AppendApiVersion(string uri, Uri startRequestUri)
+        {
+            var apiVersion = startRequestUri.Query.Trim('?').Split('&').Select(s => s.Split('=')).Where(arr => arr[0].Equals("api-version")).Select(arr => arr[1]).FirstOrDefault();
```
Comment (annelo-msft) [inline src/assets/Generator.Shared/NextLinkOperationImplementation.cs:120]: https://docs.microsoft.com/en-us/dotnet/api/system.web.httputility.parsequerystring?view=net-6.0 ?

**Review by annelo-msft — APPROVED** [review]
@AlexanderSher, I am approving this as a fix for the Cognitive library issue -- I will start keeping track of it as something we can refactor to use generated code as I proposed elsewhere.

**Comment by AlexanderSher** [comment]
Conversation goes on in https://github.com/Azure/azure-sdk-for-net/issues/25411, so I think we should wait with merge here.

**Pull request closed without merging**